src/httputil.rs

```diff
@@ -1,9 +1,20 @@
 use std::borrow::Cow;
+use std::convert::Infallible;
 
-use http::{Response, StatusCode};
-use hyper::Body;
+use http::StatusCode;
+use serde::Serialize;
 use warp::reject::Reject;
-use warp::Rejection;
+use warp::{Rejection, Reply};
+
+#[derive(Serialize, Debug)]
+pub struct Empty {}
+
+#[derive(Serialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct Error {
+    pub code: String,
```

Comment:
Are we ever going to have a non-bijective mapping between HTTP status codes and these strings?

Reply:
I suspect that we may not have a use case for such a distinction. For example, in the case of malformed vs semantically invalid queries, will the handling on the clients be any different? I suspect that a lot of errors will end up as something that gets corrected by the client's programmer in the code once they encounter it, and for most of others the handling will be to just show the error message to the user.

Reply:
That's definitely possible. Would you prefer a 
```json
{
    "error": {
        "message": "human readable error message",
    }
}
``` 
error format, that drops `code` for now, assuming http status is sufficient for machine purposes?

Reply:
Yes.

Reply:
Changed to this.